spring-integration-core/src/main/java/org/springframework/integration/context/IntegrationContextUtils.java

```diff
@@ -212,4 +215,26 @@ public static Properties getIntegrationProperties(BeanFactory beanFactory) {
 		return properties;
 	}
 
+	/**
+	 * Return a {@link BeanDefinition} with the given name,
+	 * obtained from the given {@link BeanFactory} or one of its parents.
+	 * @param name the bean name to return
+	 * @param beanFactory the {@link ConfigurableListableBeanFactory} to travers.
+	 * @return the {@link BeanDefinition} for a given name
+	 * @throws NoSuchBeanDefinitionException if a {@link BeanDefinition} is not found
+	 * @since 5.1.10
+	 */
+	public static BeanDefinition getBeanDefinition(String name, ConfigurableListableBeanFactory beanFactory) {
+		try {
+			return beanFactory.getBeanDefinition(name);
+		}
+		catch (NoSuchBeanDefinitionException ex) {
+			BeanFactory parentBeanFactory = beanFactory.getParentBeanFactory();
+			if (parentBeanFactory instanceof ConfigurableListableBeanFactory) {
```

Comment:
Why just one parent? Don't we need to recurse up the BF tree?

Reply:
Because the next `return getBeanDefinition(name, (ConfigurableListableBeanFactory) parentBeanFactory);` call is indeed recursive.

Not sure what am I missing...

Reply:
Doh - not enough coffee.